tests/phpunit/integration/Core/Authentication/Setup_Test.php

```diff
@@ -206,7 +207,73 @@ function( $preempt, $args, $url ) use ( $context, &$proxy_server_requests, $has_
 		} catch ( WPDieException $exception ) {
 			$error = $has_credentials ? 'Test error message.' : 'test_error_code';
 			$this->assertStringContainsString(
-				sprintf( 'The request to the authentication proxy has failed with an error: %s', $error ),
+				sprintf(
+					'The request to the authentication proxy has failed with an error: %1$s %2$s.',
+					$error,
+					$oauth_proxy_failed_help_link
+				),
+				$exception->getMessage()
+			);
+		}
+
+		$this->assertCount( 1, $proxy_server_requests );
+	}
+
+	/**
+	 * @dataProvider data_conditionally_syncs_site_fields
+	 */
+	public function test_handle_action_setup_start__invalid_url( $has_credentials ) {
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$context                      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE, new MutableInput() );
+		$setup                        = new Setup( $context, new User_Options( $context ), new Authentication( $context ) );
+		$oauth_proxy_failed_help_link = $setup->get_oauth_proxy_failed_help_link();
+		$setup->register();
+
+		if ( $has_credentials ) {
+			$this->fake_proxy_site_connection();
+		}
+
+		$_GET['code']  = 'test-code';
+		$_GET['nonce'] = wp_create_nonce( Google_Proxy::ACTION_SETUP_START );
+
+		$proxy_server_requests = array();
+
+		add_filter(
+			'pre_http_request',
+			function( $preempt, $args, $url ) use ( $context, &$proxy_server_requests, $has_credentials ) {
+				if ( ( new Google_Proxy( $context ) )->url( Google_Proxy::OAUTH2_SITE_URI ) !== $url ) {
+					return $preempt;
+				}
+
+				// Collect any HTTP requests to the proxy server to register/sync site with the proxy server.
+				$proxy_server_requests[] = $args;
+
+				// An invalid redirect URL (without protocol).
+				$redirect_url = 'sitekit.withgoogle.com/test-page';
+
+				return array(
+					'response' => array( 'code' => 200 ),
+					'headers'  => array( Google_Proxy::HEADER_REDIRECT_TO => $redirect_url ),
+					'body'     => '{}',
+				);
+			},
+			10,
+			3
+		);
+
+		try {
+			do_action( 'admin_action_' . Google_Proxy::ACTION_SETUP_START );
+			$this->fail( 'Expected WPDieException!' );
+		} catch ( RedirectException $redirect ) {
+			$this->fail( 'Expected WPDieException!' );
+		} catch ( WPDieException $exception ) {
+			$this->assertStringContainsString(
+				sprintf(
+					'The request to the authentication proxy has failed. Please, try again later. %s.',
+					$oauth_proxy_failed_help_link
+				),
```

Comment:
And again, test for the literal string here...
```suggestion
				'The request to the authentication proxy has failed. Please, try again later. <a href="https://sitekit.withgoogle.com/support/?error_id=request_to_auth_proxy_failed" target="_blank">Get Help</a>.',
```